### PR TITLE
feat(validate): warn on inert node and edge attributes

### PR DESCRIPTION
**Pull request by brynary**
## Summary

A declarative format must not accept attributes that do nothing. This adds two lint rules that make silently-ignored attributes warn at validate time, and reconciles the `loop_restart` docs with actual executor behavior.

### New rule: `inert_attribute`

Warns when a handler-specific attribute appears on a node type that never reads it:

| Attribute | Read only by |
|---|---|
| `script`, `language` | command |
| `duration` | wait |
| `join_policy`, `max_parallel` | parallel |
| `output_schema` | agent, prompt |
| `prompt` | agent, prompt, parallel.fan_in (LLM-judged merge) |

Deliberately excluded to avoid false positives: multi-handler attrs (`timeout` — human + command), attrs resolved for every node by the fidelity lifecycle (`fidelity`, `retry_policy`, `max_visits`, `goal_gate`), and everything a model stylesheet can inject onto any node (`model`, `provider`, `reasoning_effort`, `speed`, `backend`).

### New rule: `parallel_branch_inert_attribute`

Warns on `fidelity`/`thread_id` on parallel branch nodes and `fork -> branch` edges. Branch dispatch goes through `dispatch_handler` directly and bypasses `FidelityLifecycle::before_node`, so branches receive the context snapshot taken when the parallel node started — per-branch fidelity is a dead letter (confirmed live: a `truncate` branch received the identical compact preamble as its default sibling). The fix hint points at the parallel node, where fidelity **does** take effect today. Node-level warnings only fire when every incoming edge comes from a parallel node, so a node also reachable through a normal edge (where the attribute is honored) stays clean.

When per-branch fidelity resolution lands (planned follow-up), the `fidelity` half of this rule gets retired; `thread_id` stays warned (branches never join conversation threads).

### Docs: `loop_restart` reconciliation

`failures.mdx` described `loop_restart` only as a failure-recovery mechanism. Actual behavior (`executor.rs`: `NextStep::LoopRestart` → `state.restart(target, Some(Context::new()))`): taking the edge restarts from the target with a **fresh, empty context**, clearing stage history/outcomes/retry counts while preserving node visit counts — on success crossings as well, which cross freely; only failure crossings are gated to `transient_infra` (`circuit_breaker.rs`). Updated `failures.mdx`, the `dot-language.mdx` edge-attribute row, and the definition-of-done example's description of its `loop_restart` self-loop. No behavior change — `loop_restart` on success-conditioned edges turned out to be meaningful, so it gets accurate docs instead of a warning.

## Test plan

- [x] Unit tests for both rules (positive, negative, and boundary cases: shapeless nodes defaulting to agent, fan-in judge prompts, branch nodes with mixed incoming edges, unknown shapes deferred to `type_known`)
- [x] `cargo nextest run --workspace` — 7025 passed
- [x] `cargo +nightly-2026-04-14 clippy --workspace --all-targets -- -D warnings` — clean
- [x] `cargo +nightly-2026-04-14 fmt --check --all` — clean
- [x] Swept the repo's `.fabro` examples and the DOT compatibility corpus — no new warnings fire

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: lib/crates/fabro-validate/src/rules/parallel_branch_inert_attribute.rs [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+                    severity: Severity::Warning,
+                    message: format!(
+                        "Node '{}' sets '{attr}', but it only runs as a parallel branch (of '{parallel_id}'), where '{attr}' is ignored: branches receive the context snapshot taken when the parallel node started",
+                        node.id,
+                    ),
+                    node_id: Some(node.id.clone()),
+                    edge: None,
+                    fix: Some(fix_message(attr, &parallel_id)),
+                    ..Diagnostic::default()
+                });
```
[the remark this answers is not in the document]
Comment (brynary) [reply lib/crates/fabro-validate/src/rules/parallel_branch_inert_attribute.rs]: Fixed in 3af433509 — the node-level diagnostic now collects all unique parallel parents (sorted) and names every one in both the message and the fix hint, with singular/plural wording. Added a regression test covering a branch node shared by two forks.

File: lib/crates/fabro-validate/src/rules/parallel_branch_inert_attribute.rs [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+use crate::{Diagnostic, LintRule, Severity};
+
+pub(super) fn rule() -> Box<dyn LintRule> {
+    Box::new(Rule)
+}
+
+/// Attributes that parallel branch execution does not resolve. Branch nodes
+/// are dispatched with a snapshot of the context taken when the parallel node
+/// started, so per-branch `fidelity` never changes what a branch sees, and
+/// branches never join conversation threads regardless of `thread_id`.
```
[the remark this answers is not in the document]
Comment (brynary) [reply lib/crates/fabro-validate/src/rules/parallel_branch_inert_attribute.rs]: Fixed in f02ee6aae. The comment and fix hint now say that a branch-level `thread_id` override is ignored while the branch inherits the thread resolved when the parallel node started. Added an exact assertion for the corrected fix text.

File: lib/crates/fabro-validate/src/rules/inert_attribute.rs [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+    }
+
+    fn apply(&self, graph: &Graph) -> Vec<Diagnostic> {
+        let mut diagnostics = Vec::new();
+        for node in graph.nodes.values() {
+            // An unknown shape or type is covered by the type_known rule; a
+            // node this rule cannot classify is skipped rather than guessed at.
+            let Some(handler) = node.handler_type() else {
+                continue;
+            };
```
[the remark this answers is not in the document]
Comment (brynary) [reply lib/crates/fabro-validate/src/rules/inert_attribute.rs:41]: Fixed in f02ee6aae. `inert_attribute` now checks the shared built-in handler predicate before evaluating handler-specific attributes, so explicit unknown/custom types are left to `type_known`. Added a regression test for an explicit custom type carrying `script`.
